src/signup/AccountInfo.tsx

```diff
@@ -0,0 +1,177 @@
+// 회원가입 페이지 - 아이디, 비밀번호 등 계정 입력 컴포넌트
+import Image from "next/image";
+import React, { useEffect, useState } from "react";
+import { useSetRecoilState } from "recoil";
+import { signupPageState } from "@shared/recoil/signup";
+import Button from "./Button";
+import CheckAuth from "./CheckAuth";
+import Input from "./Input";
+import Instructions from "./Instructions";
+
+interface FormData {
+  [id: string]: { essential: boolean; value: string };
```

Comment:
[Lv2] 이 속상만 따로 타입을 지정해준 이유가 있나요? 아래 isDataComplete때문만이라면 코멘트를 참고해주시고 간결하게 정리가 가능할것같습니다!

Reply:
`[id: string]` 으로 적어주지 않으면 실행에는 오류가 없지만 ts에서 같은 파일 내의 handleInputChange 함수에서 아래와 같은 오류가 나옵니다!
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'FormData'.
  No index signature with a parameter of type 'string' was found on type 'FormData'.
```
FormData의 index 타입이 string이 아니라서 그런 것 같아서, 따로 타입을 지정해주니 오류가 사라지긴 했습니다.
혹시 이 부분은 다른 해결 방법이 있을까요?

Reply:
흠.. 이경우엔 타입단언을 해주는게 맞을것같습니다. 위에 FormData Key들을 지칭하는 변수를 만들고 그걸로 단언을 해주면 좋을것같아요

예시
```suggestion
type FormData = Record<FormKey, FormValue>;
type FormValue = { essential: boolean; value: string };
type FormKey = "id" | "password" | "checkPassword" | "phone";

...

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setFormData((prevFormData) => ({
      ...prevFormData,
      [name]: { ...prevFormData[name as FormKey], value },
    }));
  };

```